File: examples/wf-versions/src/main/java/io/littlehorse/examples/MyWorker.java
package io.littlehorse.examples;

import io.littlehorse.sdk.worker.LHTaskMethod;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class MyWorker {

    private static final Logger log = LoggerFactory.getLogger(MyWorker.class);

    @LHTaskMethod("greet0")
    public String greeting0(String name) {
        log.debug("Executing task greet0");
        return "hello there, " + name;
    }

    @LHTaskMethod("greet1")
    public String greeting1(String name) {
        log.debug("Executing task greet1");
        return "hello there, " + name + ", this is the latest version";
    }
}

File: sdk-dotnet/Examples/ConditionalsExample/MyWorker.cs
using LittleHorse.Sdk.Worker;

namespace ConditionalsExample;

public class MyWorker
{
    [LHTaskMethod("task-a")]
    public async Task<string> TaskA() {
        Console.WriteLine("Executing task-a");
        return "hello there A";
    }

    [LHTaskMethod("task-b")]
    public async Task<string> TaskB() {
        Console.WriteLine("Executing task-b");
        return "hello there B";
    }

    [LHTaskMethod("task-c")]
    public async Task<string> TaskC() {
        Console.WriteLine("Executing task-c");
        return "hello there C";
    }

    [LHTaskMethod("task-d")]
    public async Task<string> TaskD() {
        Console.WriteLine("Executing task-d");
        return "hello there D";
    }
}